doc: document `spec.csi` section in cephcluster crd (#13375)
rook/rook: Documentation/CRDs/Cluster/ceph-cluster-crd.md

    user's responsibility.

The operator does not unset any removed config options, it is the user's responsibility to unset or set the default value for each removed option manually using the Ceph CLI.

## CSI Driver Options

> The CSI driver options mentioned here are applied per ceph cluster. The following options are available:

Reviewer: ```suggestion
The CSI driver options mentioned here are applied per Ceph cluster. The following options are available:
```

> The CSI driver options mentioned here are applied per Ceph cluster. The following options are available:

Reviewer 2: per Ceph cluster(external or converged)

Reviewer 3: I think no need to mention external/converged. There is nothing special for csi settings between those two modes.

> * `readAffinity`: Refer to the read affinity section in [ceph-csi-drivers.md](../../Storage-Configuration/Ceph-CSI/ceph-csi-drivers.md#enable-read-affinity-for-rbd-and-cephfs-volumes) for more details.

Reviewer 3: In the text link we want to use the normal text, instead of a filename. Also, a brief explanation is helpful.
```suggestion
* `readAffinity`: RBD volumes allow serving reads from an OSD in proximity to the client. Refer to the read affinity section in the [Ceph CSI Drivers](../../Storage-Configuration/Ceph-CSI/ceph-csi-drivers.md#enable-read-affinity-for-rbd-and-cephfs-volumes) doc for more details.
```

> * `readAffinty`: This section contains options related to read affinity for the CSI driver. Refer to the [read affinity](../../Storage-Configuration/Ceph-CSI/ceph-csi-drivers.md#enable-read-affinity-for-rbd-and-cephfs-volumes) for more details.

Reviewer: ```suggestion
* `readAffinity`: This section contains options related to reading affinity for the CSI driver. Refer to the [read affinity](../../Storage-Configuration/Ceph-CSI/ceph-csi-drivers.md#enable-read-affinity-for-rbd-and-cephfs-volumes) for more details.
```

    * `enabled`: Whether to enable read affinity for the CSI driver. Default is `false`.

>     * `crushLocationLabels`:  Defines which node labels to use as CRUSH location. This should correspond to the values set in the CRUSH map. Defaults to the labels mentioned in the

Reviewer 3: ```suggestion
    * `crushLocationLabels`:  Node labels to use as CRUSH location, corresponding to the values set in the CRUSH map. Defaults to the labels mentioned in the
```

> [OSD topology](../../CRDs/Cluster/ceph-cluster-crd.md#osd-topology) topic.

Reviewer 3: In the same doc we can simplify the relative path
```suggestion
[OSD topology](ceph-cluster-crd.md#osd-topology) topic.
```

* `cephfs`:

>     * `kernelMountOptions`: Defines the mount options for kernel mounter. Refer to the [kernel mount options](https://docs.ceph.com/en/latest/man/8/mount.ceph/#options) for more details.

Reviewer 3: ```suggestion
    * `kernelMountOptions`: Mount options for kernel mounter. Refer to the [kernel mount options](https://docs.ceph.com/en/latest/man/8/mount.ceph/#options) for more details.
```

>     * `kernelMountOptions`: KernelMountOptions defines the mount options for kernel mounter. Refer to the [kernel mount options](https://docs.ceph.com/en/latest/man/8/mount.ceph/#options) for more details.

Reviewer 3: ```suggestion
    * `kernelMountOptions`: Defines the mount options for kernel mounter. Refer to the [kernel mount options](https://docs.ceph.com/en/latest/man/8/mount.ceph/#options) for more details.
```

>     * `fuseMountOptions`: Defines the mount options for fuse mounter. Refer to the [fuse mount options](https://docs.ceph.com/en/latest/man/8/ceph-fuse/#options) for more details.

Reviewer 3: No need for "Defines the...", for consistency with other rook docs
```suggestion
    * `fuseMountOptions`: Mount options for fuse mounter. Refer to the [fuse mount options](https://docs.ceph.com/en/latest/man/8/ceph-fuse/#options) for more details.
```

>     * `fuseMountOptions`: FuseMountOptions defines the mount options for fuse mounter. Refer to the [fuse mount options](https://docs.ceph.com/en/latest/man/8/ceph-fuse/#options) for more details.

Reviewer 3: ```suggestion
    * `fuseMountOptions`: Defines the mount options for fuse mounter. Refer to the [fuse mount options](https://docs.ceph.com/en/latest/man/8/ceph-fuse/#options) for more details.
```
